### PR TITLE
Cluster version

### PR DESCRIPTION
**Pull request by jim-minter**
add a cluster version field
add a resource group field and make it configurable

### CODE REVIEW

File: pkg/api/v20191231preview/openshiftcluster_validatestatic.go [hunk 1]
```diff
@@ -126,6 +127,25 @@ func (sv *openShiftClusterStaticValidator) validateProperties(path string, p *Pr
 	return nil
 }
 
+func (sv *openShiftClusterStaticValidator) validateClusterProfile(path string, cp *ClusterProfile) error {
+	if !rxDomainName.MatchString(cp.Domain) {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".domain", "The provided domain '%s' is invalid.", cp.Domain)
+	}
+	switch cp.Version {
+	case "", "4.3.0-0.nightly-2019-12-05-001549":
```
Comment (mjudeikis) [inline pkg/api/v20191231preview/openshiftcluster_validatestatic.go]: I expect this to change in the future (?)

File: pkg/backend/openshiftcluster/create.go [hunk 2]
```diff
@@ -203,8 +205,12 @@ func (m *Manager) Create(ctx context.Context) error {
 		return err
 	}
 
-	image := &releaseimage.Image{
-		PullSpec: "arosvc.azurecr.io/openshift-release-dev/ocp-release-nightly@sha256:5f1ff5e767acd58445532222c38e643069fdb9fdf0bb176ced48bc2eb1032f2a",
+	image := &releaseimage.Image{}
```
Comment (mjudeikis) [inline pkg/backend/openshiftcluster/create.go]: Is the plan in the future to call private Cincinnati three?

File: pkg/frontend/openshiftcluster_putorpatch.go [hunk 3]
```diff
@@ -130,7 +129,9 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 	doc.OpenShiftCluster.ID, doc.OpenShiftCluster.Name, doc.OpenShiftCluster.Type = oldID, oldName, oldType
 
 	if isCreate {
+		doc.ClusterResourceGroupIDKey = strings.ToLower(doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
 		doc.OpenShiftCluster.Properties.ProvisioningState = api.ProvisioningStateCreating
+		doc.OpenShiftCluster.Properties.ClusterProfile.Version = "4.3.0-0.nightly-2019-12-05-001549"
```
Comment (mjudeikis) [inline pkg/frontend/openshiftcluster_putorpatch.go]: ?

**Comment by mjudeikis** [comment]
good to merge :)

File: pkg/api/v20191231preview/openshiftcluster_validatestatic.go (hunk 1 above)
Comment (jim-minter) [reply pkg/api/v20191231preview/openshiftcluster_validatestatic.go]: yes

File: pkg/backend/openshiftcluster/create.go (hunk 2 above)
Comment (jim-minter) [reply pkg/backend/openshiftcluster/create.go]: perhaps we will let the user choose within a range of versions that they can deploy

File: pkg/frontend/openshiftcluster_putorpatch.go (hunk 3 above)
Comment (jim-minter) [reply pkg/frontend/openshiftcluster_putorpatch.go]: this is where we initialise the cluster version that is going to be deployed.  Until it is a nice simple string like "4.3.0" that the user can choose, I'm hard-coding it

**Pull request merged**